sklearn/isotonic.py

```diff
@@ -252,7 +278,7 @@ def _build_f(self, X, y):
             # single y, constant prediction
             self.f_ = lambda x: y.repeat(x.shape)
         else:
-            self.f_ = interpolate.interp1d(X, y, kind='slinear',
+            self.f_ = interpolate.interp1d(X, y, kind='linear',
```

Comment:
That makes sense @ogrisel, I couldn't find exactly the same dataset I was having trouble with back in version 0.15. I used this branch's implementation in my newer datasets and it seems to work fine. Indeed the _make_unique function seems to have fixed the problems, I tried to remove it and I got the same errors I was getting before. :+1: